main.tf

```diff
@@ -0,0 +1,293 @@
+locals {
+  rules_string = flatten([
+    for key, rule in local.sorted_fqdn_rules : [
+      for id, fqdn in rule.fqdns : {
+        http = format("%s http $src_ip_%d any -> any $HTTP_PORTS (http.host;dotprefix;content:\"%s\";endswith;flow:to_server,established;sid:%d%d2;)", lower(rule.action), key, fqdn, key, id)
+        tls  = format("%s tls $src_ip_%d any -> any $HTTP_PORTS (tls.sni;dotprefix;content:\"%s\";endswith;nocase;flow:to_server,established;sid:%d%d1;)", lower(rule.action), key, fqdn, key, id)
+    }]
+  ])
+
+  sorted_fqdn_rules = values({
+    for k, v in var.fqdn_rules : format("%03d", v.priority) => {
+      action      = v.action
+      description = k
+      fqdns       = v.fqdns
+      priority    = v.priority
+      source_ip   = v.source_ip
+    }
+  })
+
+  sorted_ip_rules = values({
+    for k, v in var.ip_rules : format("%03d", v.priority) => {
+      action           = v.action
+      description      = k
+      destination_ip   = v.destination_ip
+      destination_port = v.destination_port
+      direction        = v.direction
+      priority         = v.priority
+      protocol         = v.protocol
+      source_ip        = v.source_ip
+      source_port      = v.source_port
+    }
+  })
+}
+
+data "aws_region" "default" {}
+
+################################################################################
+# AWS Network Firewall
+################################################################################
+
+resource "aws_networkfirewall_firewall" "default" {
```

Comment:
add variable with deletion protection, set default to true. 
add `encryption_configuration`, require passing a KMS key